app/apphandlers/embeddedAppHandler.go

```diff
@@ -200,6 +202,20 @@ func (a *embeddedApp) StartRudderCore(ctx context.Context, options *app.Options)
 		jobsdb.WithDSLimit(&a.config.processorDSLimit),
 		jobsdb.WithSkipMaintenanceErr(config.GetBool("Processor.jobsDB.skipMaintenanceError", false)),
 	)
+	defer schemaDB.Close()
+
+	archivalDB := jobsdb.NewForReadWrite(
+		"arc",
+		jobsdb.WithClearDB(options.ClearDB),
+		jobsdb.WithDSLimit(&a.config.processorDSLimit),
+		jobsdb.WithSkipMaintenanceErr(config.GetBool("Processor.jobsDB.skipMaintenanceError", false)),
+		jobsdb.WithJobMaxAge(
```

Comment:
**Note for future improvement:** `jobsdb.HandleT#doCleanup` method picks up jobs in **executing** state as well.

This means that both the following are theoretically possible:
- Duplicate terminal states are recorded for a job, one by the cleanup goroutine and another by the goroutine which marked jobs as executing.
- The cleanup goroutine marks a job as aborted and the other goroutine marks it as failed.

Thus we might need to reconsider in the future having the following in mind:
1. Dealing with jobs in executing state
2. Keeping pending events consistent